components/inputs/input-date.js

```diff
@@ -1,11 +1,24 @@
+import '../button/button-subtle.js';
+import '../dropdown/dropdown.js';
+import '../dropdown/dropdown-content.js';
+import '../focus-trap/focus-trap.js';
 import '../icons/icon.js';
 import './input-text.js';
 import { css, html, LitElement } from 'lit-element/lit-element.js';
-import { formatDate, parseDate } from '@brightspace-ui/intl/lib/dateTime.js';
+import { formatDate, getDateTimeDescriptor, parseDate } from '@brightspace-ui/intl/lib/dateTime.js';
+import { getToday } from '../calendar/calendar.js';
 import { ifDefined } from 'lit-html/directives/if-defined.js';
 import { LocalizeStaticMixin } from '../../mixins/localize-static-mixin.js';
 import { RtlMixin } from '../../mixins/rtl-mixin.js';
 
+let calendarDataDescriptor;
```

Comment:
I'm thinking that some of these functions that overlap between `calendar` and `input-date` should be pulled out into a separate helper

Reply:
So I think I will leave the caching related to `dateTimeDescriptor` within core for now, as it seems a bit complicated to move it into intl and be able to keep up to date when language/locale settings changes, and I'd like to do that more separately.